src/bp/core/services/dialog/dialog-engine.ts

```diff
@@ -461,12 +517,20 @@ export class DialogEngine {
     return this.processEvent(sessionId, event)
   }
 
-  private async _goToSubflow(botId: string, event: IO.IncomingEvent, sessionId: string, parentFlow, parentNode) {
-    const subflowName = parentNode.flow // Name of the subflow to transition to
+  private async _goToSubflow(
+    botId: string,
+    event: IO.IncomingEvent,
+    sessionId: string,
+    parentFlow,
+    parentNode: FlowNode
+  ) {
+    const subflowName = parentNode.flow!
     const subflow = this._findFlow(botId, subflowName)
     const subflowStartNode = this._findNode(botId, subflow, subflow.startNode)
 
+    event.state.session.nduContext!.last_topic = parseFlowName(subflowName).topic!
```

Comment:
I'm setting the current topic for the ndu context manually here because it's otherwise impossible to change it unless going through a trigger of type `workflow`.

![Capture2](https://user-images.githubusercontent.com/59894025/89825140-25eecf00-db22-11ea-9963-a1f29af513db.PNG)

Right now any `jumpTo` or `transitionTo` logic going into another topic won't work with the ndu because they don't change `last_topic` (the transition will still happen but the ndu will believe that is in the wrong topic).

I'm not sure why `last_topic` is kept as an independent variable. Couldn't we just use context.currentFlow so we take our current topic from only one location (I actually tried doing that but it broke a lot stuff and I don't understand why).

![Capture](https://user-images.githubusercontent.com/59894025/89825131-225b4800-db22-11ea-9f92-2fb14c65c2cf.PNG)

Reply:
yes, can you please add a TODO to revisit this part of the NDU and also to remove your hack line 531?  it's a bit more complex because we have to keep track of workflows that ended (thus currentFlow is unset)